runtime/src/bank.rs

```diff
@@ -1210,16 +1211,66 @@ impl Bank {
         }
     }
 
+    fn distribute_rent_to_validators(
+        &self,
+        vote_account_hashmap: &HashMap<Pubkey, (u64, Account)>,
+        rent_to_be_distributed: u64,
+    ) {
+        let mut node_stake_hashmap: HashMap<Pubkey, u64> = HashMap::new();
+        let mut total_staked = 0;
+
+        vote_account_hashmap
+            .iter()
+            .map(|(_vote_pubkey, (staked, account))| {
+                total_staked += *staked;
+                VoteState::deserialize(&account.data)
+                    .ok()
+                    .map(|vote_state| (*staked, vote_state.node_pubkey))
+                    .unwrap_or((0, Pubkey::default()))
+            })
+            .collect::<HashMap<u64, Pubkey>>()
```

Comment:
you don't need to sum the stakes before distribution, just distribute according to stake % in this iterator

Reply:
@rob-solana So, if there is 1:1 relation between node pubkey and vote account (which node uses to vote), what's the reasoning behind  this additional indirection? Is it because we want to keep node private key out of picture and safe?

Reply:
it's just the way vote accounts evolved.  we need:
1. (original requirement) for the voting keypair to be separate from the node keypair because voting can cause slashing, and the node keypair is used *a lot* for gossip
2. to be able to re-assign the voter keypair
3. to be able to pay fees from the validator's account
4. (and a new requirement) to be able to re-assign the validator

Reply:
cool. Made changes.

Reply:
@rob-solana Can you expand on 1st requirement? Are you saying here that, by separating voting keypair we are able to keep track of signing done by voting keypair (and so, avoiding double signing in case of restart/crash etc etc), which would have been difficult if those two are same and used for multiple purpose.